test/k8pshTest.cxx

```diff
@@ -0,0 +1,135 @@
+
+	int exitCode = WEXITSTATUS(status);
+#endif
+
+	std::cout << "Command \"" << command << "\" returned " << exitCode << std::endl;
+	return exitCode;
+}
+
+int main(int argc, const char *argv[])
+{
+	std::string executable = k8psh::Utilities::getExecutablePath();
+	std::string basename = k8psh::Utilities::getBasename(argv[0]);
+
+	// Check for test cases
+	if (argc > 1)
+	{
+		std::string arg = argv[1];
+
+		if (arg == "--name")
+			return mainServer(argc, argv);
+		else if (arg == "0")
+		{
+			TEST_THAT(k8psh::Utilities::getEnvironmentVariable("PATH") == ".");
+			std::cout << k8psh::Utilities::getEnvironmentVariable("K8PSH_TEST_NAME") << std::endl;
+			return 0;
+		}
+		else if (arg == "1")
+		{
+			if (argc < 3)
+				LOG_ERROR << "Expecting additional argument to command";
+
+			TEST_THAT(k8psh::Utilities::getEnvironmentVariable("PATH") != ".");
+			std::cerr << k8psh::Utilities::getEnvironmentVariable("K8PSH_TEST_NAME") << ", " << argv[2] << std::endl;
+			return 1;
+		}
+		else if (arg == "2")
+		{
+			TEST_THAT(k8psh::Utilities::getEnvironmentVariable("PATH") == ".");
+			std::cout << std::cin.rdbuf();
+
+			return 2;
+		}
+		else if (arg == "3")
+		{
+			TEST_THAT(k8psh::Utilities::getEnvironmentVariable("PATH") != ".");
+			TEST_THAT(std::fclose(stdin) == 0);
+			TEST_THAT(std::fclose(stdout) == 0);
+			TEST_THAT(std::fclose(stderr) == 0);
+			std::this_thread::sleep_for(std::chrono::milliseconds(250));
+
+			return 3;
+		}
+
+		return mainClient(argc, argv);
+	}
+	else if (basename != "k8pshTest" && basename != "k8pshTest.exe")
+		return mainClient(argc, argv);
+
+	// Create configuration file
+	std::ofstream configFile((basename + ".conf").c_str());
+
+	configFile << "workingDirectory = ." << std::endl;
+	configFile << "[k8pshTest]" << std::endl;
+	configFile << "'0_" << basename << "' K8PSH_TEST_NAME= PATH= '" << executable << "' 0" << std::endl;
+	configFile << "'1_" << basename << "' K8PSH_TEST_NAME= =PATH= '" << executable << "' 1" << std::endl;
+	configFile << "'2_" << basename << "' K8PSH_TEST_NAME= ?PATH= '" << executable << "' 2" << std::endl;
+	configFile << "'3_" << basename << "' K8PSH_TEST_NAME= ?PATH= '" << executable << "' 3" << std::endl;
+	configFile.close();
+
+	// Start server
+	k8psh::Utilities::setEnvironmentVariable("K8PSH_CONFIG", basename + ".conf");
+
+	std::thread([executable]{ (void)runCommand(executable + " --name k8pshTest --generate-local-executables --timeout 3000"); }).detach();
+	std::this_thread::sleep_for(std::chrono::milliseconds(250));
+
+	// Run test cases
+	int exitCode = 0;
+
+	try
+	{
+		TEST_THAT(k8psh::Utilities::setEnvironmentVariable("PATH", "."));
+		TEST_THAT(k8psh::Utilities::setEnvironmentVariable("K8PSH_TEST_NAME", "Test 0"));
+		TEST_THAT(runCommand("0_" + basename + " > test.out 2> test.err") == 0);
+		TEST_THAT(k8psh::Utilities::readFile("test.out") == static_cast<std::ostringstream &>(std::ostringstream() << "Test 0" << std::endl).str());
```

Comment:
Line endings are translated when sent to stdout/stderr but not in stringstreams, so they will not match on Windows. Need to either remove line endings or come up with different solution.